api/envoy/config/core/v3/address.proto

```diff
@@ -32,15 +32,19 @@ message Pipe {
 }
 
 // The address represents an envoy internal listener.
-// [#comment: TODO(lambdai): Make this address available for listener and endpoint.
 // TODO(asraa): When address available, remove workaround from test/server/server_fuzz_test.cc:30.]
 message EnvoyInternalAddress {
   oneof address_name_specifier {
     option (validate.required) = true;
 
-    // [#not-implemented-hide:] The :ref:`listener name <envoy_v3_api_field_config.listener.v3.Listener.name>` of the destination internal listener.
+    // Specifies the :ref:`name <envoy_v3_api_field_config.listener.v3.Listener.name>` of the
+    // internal listener.
     string server_listener_name = 1;
   }
+
+  // Specifies an endpoint identifier to distinguish between multiple endpoints for the same internal listener in a
+  // single upstream pool. Only used in the upstream addresses.
```

Comment:
Is empty string the default value here, or some unique name for the endpoint?
Is there a requirement that each endpoint id will be unique?

Reply:
Yes, the empty string is the default. There's no requirement per-se, but it's de-duplicated in the same way as when EDS has duplicate IP addresses.

Reply:
Ah, I see. I think it may be beneficial to state that this is used for tracking purposes (opposed to an ID that will be used as a foreign key).

Reply:
Done.